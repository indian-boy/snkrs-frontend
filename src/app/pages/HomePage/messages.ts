import { translations } from 'locales/translations';
import { _t } from 'utils/messages';

export const messages = {
  i18nTitle: () => _t(translations.homePage.pageTitle, 'Home Page'),
  i18nPageMetaContent: () =>
    _t(translations.homePage.pageMetaContent, 'Home Page - Meta Content'),
  i18nShoppingStores: () => _t(translations.homePage.shoppingStores, 'Lojas'),
  i18nSearchPlaceholder: () =>
    _t(translations.homePage.searchPlaceholder, 'Busque por endereço ou CEP'),
  i18nSearchButtonLabel: () =>
    _t(translations.homePage.searchButtonLabel, 'Buscar'),
  i18nMinorDistance: () =>
    _t(translations.homePage.filterMinorDistanceLabel, 'Menor distância'),
  i18nGreaterDistance: () =>
    _t(translations.homePage.filterGreaterDistance, 'Maior distância'),
};
