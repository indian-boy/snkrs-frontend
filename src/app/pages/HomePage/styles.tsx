import styled from 'styled-components/macro';

const Wrapper = styled.div`
  padding: 2rem 1rem;
  background: ${props => props.theme.palette.secondary.default};
  display: flex;
  justify-content: center;
`;

const Title = styled.span`
  color: ${props => props.theme.palette.primary.p700};
  font-weight: bold;
  font-size: 2rem;
  line-height: 1.2rem;
`;

const Main = styled.main`
  display: flex;
  flex-direction: column;
  gap: 2rem;
  align-items: center;
`;

export { Wrapper, Title, Main };
