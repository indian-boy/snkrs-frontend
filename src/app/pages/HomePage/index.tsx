import * as React from 'react';
import { Helmet } from 'react-helmet-async';
import { useTranslation } from 'react-i18next';

import { messages } from './messages';

export function HomePage() {
  const { t } = useTranslation();

  return (
    <>
      <Helmet>
        <title>{t(messages.i18nTitle())}</title>
        <meta name="description" content={t(messages.i18nPageMetaContent())} />
      </Helmet>
      <span>HomePage container</span>
    </>
  );
}
