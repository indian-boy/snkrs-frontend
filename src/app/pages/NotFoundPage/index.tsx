import React from 'react';
import { Helmet } from 'react-helmet-async';
import { useTranslation } from 'react-i18next';
import styled from 'styled-components/macro';
import { messages } from './messages';

export function NotFoundPage() {
  const { t } = useTranslation();

  return (
    <>
      <Helmet>
        <title>{t(messages.i18nPageTitle())}</title>
        <meta name="description" content={t(messages.i18nPageMetaContent())} />
      </Helmet>
      <Wrapper>
        <Title>
          4
          <span role="img" aria-label="Crying Face">
            ðŸ˜¢
          </span>
          4
        </Title>
        <Message>{t(messages.i18nMessage())}</Message>
      </Wrapper>
    </>
  );
}

const Wrapper = styled.div`
  height: 100vh;
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  min-height: 20rem;
`;

const Title = styled.div`
  margin-top: -8vh;
  font-weight: bold;
  color: black;
  font-size: 3.375rem;

  span {
    font-size: 3.125rem;
  }
`;

export const Message = styled.p`
  font-size: 1rem;
  line-height: 1.5;
  color: black;
  margin: 0.625rem 0 1.5rem 0;
`;
