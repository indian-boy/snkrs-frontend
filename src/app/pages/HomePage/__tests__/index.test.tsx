import * as React from 'react';
import { createRenderer } from 'react-test-renderer/shallow';
import { HomePage } from 'app/pages/HomePage';

const renderer = createRenderer();

jest.mock('react-i18next', () => ({
  useTranslation: () => {
    return {
      t: str => str,
      i18n: {
        changeLanguage: () => new Promise(() => {}),
      },
    };
  },
}));

describe('<HomePage />', () => {
  it('should render and match the snapshot', () => {
    renderer.render(<HomePage />);
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});
