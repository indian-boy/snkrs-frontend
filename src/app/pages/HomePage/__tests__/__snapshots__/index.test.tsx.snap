// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> should render and match the snapshot 1`] = `
<React.Fragment>
  <Helmet
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      homePage.pageTitle
      Home Page
    </title>
    <meta
      content={
        Array [
          "homePage.pageMetaContent",
          "Home Page - Meta Content",
        ]
      }
      name="description"
    />
  </Helmet>
  <styles__Wrapper>
    <styles__Main>
      <styles__Title>
        homePage.shoppingStores
        Lojas
      </styles__Title>
      <styles__SearchWrapper>
        <styles__SearchInputCustomized
          isRounded={true}
          onChange={[Function]}
          placeholder={
            Array [
              "homePage.searchPlaceholder",
              "Busque por endereÃ§o ou CEP",
            ]
          }
          role="homeSearchInput"
          showSearchIcon={true}
          type="search"
        />
        <Memo
          label={
            Array [
              "homePage.searchButtonLabel",
              "Buscar",
            ]
          }
          onClick={[Function]}
          role="homeSearchButton"
        />
      </styles__SearchWrapper>
    </styles__Main>
    <Unknown />
  </styles__Wrapper>
</React.Fragment>
`;
